core/src/main/scala/io/finch/Input.scala

```diff
@@ -10,6 +13,48 @@ final case class Input(request: Request, path: Seq[String]) {
   def headOption: Option[String] = path.headOption
   def drop(n: Int): Input = copy(path = path.drop(n))
   def isEmpty: Boolean = path.isEmpty
+
+  def withBody(buf: Buf, charset: Option[Charset] = None): Input = {
+    request.content = buf
+    request.contentLength = buf.length.toLong
+    if (charset.isDefined) request.charset = charset.get.displayName()
+    this
+  }
+
+  def withHeaders(headers: (String, String)*): Input = {
+    headers.foreach { case (k, v) => request.headerMap.set(k, v) }
+    this
+  }
+
+  def withForm(params: (String, String)*): Input = {
+    require(request.host.isDefined, "The host has to be defined")
+    require(params.nonEmpty, "Cannot create request without params")
+    val req = RequestBuilder()
+      .addFormElement(params: _*)
+      .url(request.host.get)
+      .buildFormPost()
+    request.method = req.method
+    request.content = req.content
+    if (req.contentLength.isDefined) request.contentLength = req.contentLength.get
+    if (req.charset.isDefined) request.charset = req.charset.get
+    this
+  }
+
+  def withMultipartForm(fileElements: FileElement*)(params: (String, String)*): Input = {
```

Comment:
How about using `c.t.f.http.exp.Multipart` here? I really don't want to use `FileElement` from the `RequestBuilder` API here, but we already use `Multipart.FileUpload` in Finch so we should be fine.